packages/ra-core/src/auth/useLogin.stories.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import usePermissions, { UsePermissionsResult } from './usePermissions';
+import { CoreAdminContext, TestMemoryRouter, useLogin, useLogout } from '..';
+
+export default {
+    title: 'ra-core/auth/useLogin',
+};
+
+const getAuthProviderWithLoginAndLogout = () => {
+    let isLoggedIn = false;
+    return {
+        login: () => {
+            isLoggedIn = true;
+            return Promise.resolve();
+        },
+        logout: () => {
+            isLoggedIn = false;
+            return Promise.resolve();
+        },
+        checkAuth: () => (isLoggedIn ? Promise.resolve() : Promise.reject()),
+        checkError: () => Promise.reject('bad method'),
+        getPermissions: () =>
+            new Promise(resolve =>
+                setTimeout(resolve, 300, isLoggedIn ? 'admin' : 'guest')
+            ),
+    };
+};
+
+const LoginButton = () => {
+    const login = useLogin();
+    return <button onClick={login}>Login</button>;
+};
+
+const LogoutButton = () => {
+    const logout = useLogout();
+    return <button onClick={logout}>Logout</button>;
+};
+
+const UsePermissions = ({
+    children,
+}: {
+    children: (state: UsePermissionsResult<any, Error>) => React.ReactNode;
+}) => {
+    const permissionQueryParams = {
+        retry: false,
+    };
+    const res = usePermissions({}, permissionQueryParams);
+    return children(res);
```

Comment:
Why use a child function here? You could just return the JSX you have in `inspectPermissionsState`, couldn't you?

Reply:
Didn't give it too much though TBH, I just copied that code from the usePermissions stories